rhombus/private/listable.rkt

```diff
@@ -0,0 +1,62 @@
+#lang racket/base
+(require (for-syntax racket/base
+                     "interface-parse.rkt")
+         "treelist.rkt"
+         "provide.rkt"
+         (submod "list.rkt" for-listable)
+         (submod "list.rkt" for-compound-repetition)
+         (submod "annotation.rkt" for-class)
+         "name-root.rkt"
+         "realm.rkt"
+         (only-in "class-desc.rkt" define-class-desc-syntax)
+         "define-arity.rkt"
+         (submod "dot.rkt" for-dot-provider)
+         "call-result-key.rkt"
+         "dot-provider-key.rkt"
+         "dot-parse.rkt")
+
+(provide (for-spaces (rhombus/class
+                      rhombus/namespace
+                      rhombus/annot)
+                     Listable))
+
+(define-class-desc-syntax Listable
+  (interface-desc #'Listable
+                  #'Listable
+                  #'()
+                  #'prop:Listable
+                  #'prop:Listable
+                  #'Listable-ref
+                  (vector-immutable (box-immutable 'to_list))
+                  #'#(#:abstract)
+                  (hasheq 'to_list 0)
+                  #hasheq()
+                  #t
+                  '()
+                  #f
+                  #'()
+                  #f
+                  '()))
+
+(define-annotation-syntax Listable
+  (identifier-annotation #'listable? #'((#%dot-provider listable-instance))))
+
+(define-dot-provider-syntax listable-instance
+  (dot-provider
+   (dot-parse-dispatch
+    (lambda (field-sym field-proc ary nary fail-k)
+      (case field-sym
+        [(to_list) (nary 1 #'to_list #'make_to_list)]
+        [else (fail-k)])))))
+
+(define-name-root Listable
+  #:fields
+  (to_list))
+
+(define/arity (to_list v)
```

Comment:
This can use `define/method`, which generates a `<name>/method`.  Also `who` can be used in the body if we change the name to `Listable.to_list`.

Reply:
HF